Debian #250438, #344822, #317612; set $HOME to /etc/X11 when starting the X
server, causing the code that looks for $HOME/xorg.conf before other files in
/etc to look at our configfile, and not one that might be in root's homedir.

XXX Clarify the two changes below; one seem to ignore a compatibility file, the
other sets HOME to avoid /root/xorg.conf, but this seems to be a xorg bug.

Index: gdm-2.20.0/daemon/server.c
===================================================================
--- gdm-2.20.0.orig/daemon/server.c	2007-09-17 16:45:50.000000000 +0200
+++ gdm-2.20.0/daemon/server.c	2007-10-06 15:00:12.229024603 +0200
@@ -1358,11 +1358,20 @@
 				   (int)d->server_uid);
 			_exit (SERVER_ABORT);
 		}
+	/*
+	 * When started as root, X will read $HOME/xorg.conf before files in /etc.  As this isn't what users
+	 * expect from gdm starting up, we set $HOME to be /etc/X11.  (an unset $HOME will cause the X server
+	 * to bail)  The Debian X server will have this removed soon, and hopefully the change will go
+	 * upstream, at which point the original code will not cause user confusion.
+	 */
+#if 0
 		if (pwent->pw_dir != NULL &&
 		    g_file_test (pwent->pw_dir, G_FILE_TEST_EXISTS))
 			g_setenv ("HOME", pwent->pw_dir, TRUE);
 		else
 			g_setenv ("HOME", "/", TRUE); /* Hack */
+#endif
+		g_setenv ("HOME", "/etc/X11", TRUE);
 		g_setenv ("SHELL", pwent->pw_shell, TRUE);
 		g_unsetenv ("MAIL");
 
